feat: OpenAPI integrated for mirror node models generation

ivaylogarnev-limechain opened the pull request:
Description:
This PR integrates `OpenAPI` for generating Mirror Node models and incorporates their usage in the code where applicable. It also adds aliases for importing these models and includes various refactors.

Related issue(s):
#276 

Checklist
- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

rwalworth reviewed (commented):
Do you think we should place instructions for generating these models somewhere in the README somewhere? Might be useful for anyone to know.

Also, is it common practice to name a directory with a `.ts` extension?

ivaylogarnev-limechain commented:
> Do you think we should place instructions for generating these models somewhere in the README somewhere? Might be useful for anyone to know.
> 
> Also, is it common practice to name a directory with a `.ts` extension?

Yes, absolutely! I initially thought we were generating these models just for our own use, but you’re right—it’s important for everyone to be aware of the process. To address this, I’ve added a detailed section to the README explaining OpenAPI model generation. Additionally, I had previously handled the post-generation cleanup manually, but I’ve now introduced a Taskfile to automate the process. 

As for the `.ts` extension in a directory name—no, that was definitely a typo :D

rwalworth reviewed (approved):
LGTM, awesome work!
